From 1317d81cae3ce153439929f2338271ab98011709 Mon Sep 17 00:00:00 2001
From: Junio C Hamano <gitster@pobox.com>
Date: Fri, 20 Dec 2024 10:02:15 -0800
Subject: [PATCH 01/28] UPSTREAM: show-index: the short help should say the
 command reads from its input

The short help text given by "git show-index -h" says

    $ git show-index -h
    usage: git show-index [--object-format=<hash-algorithm>]

        --[no-]object-format <hash-algorithm>
                              specify the hash algorithm to use

The command takes a pack .idx file from its standard input.  The
user has to _know_ this, as there is no indication from this output.

Give a hint that the data to work on is fed from its standard input.

Signed-off-by: Junio C Hamano <gitster@pobox.com>

(cherry picked from commit 49edce4ff96d003e2b50202105107ef42e1f730e)
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 Documentation/git-show-index.txt | 2 +-
 builtin/show-index.c             | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Documentation/git-show-index.txt b/Documentation/git-show-index.txt
index e49318a5a..7e574ea24 100644
--- a/Documentation/git-show-index.txt
+++ b/Documentation/git-show-index.txt
@@ -9,7 +9,7 @@ git-show-index - Show packed archive index
 SYNOPSIS
 --------
 [verse]
-'git show-index' [--object-format=<hash-algorithm>]
+'git show-index' [--object-format=<hash-algorithm>] < <pack-idx-file>
 
 
 DESCRIPTION
diff --git a/builtin/show-index.c b/builtin/show-index.c
index 756d632b5..9d4ecf5e7 100644
--- a/builtin/show-index.c
+++ b/builtin/show-index.c
@@ -9,7 +9,7 @@
 #include "parse-options.h"
 
 static const char *const show_index_usage[] = {
-	"git show-index [--object-format=<hash-algorithm>]",
+	"git show-index [--object-format=<hash-algorithm>] < <pack-idx-file>",
 	NULL
 };
 
-- 
2.48.1

